src/data/polynomial/degree/definitions.lean

```diff
@@ -503,6 +503,15 @@ lemma nat_degree_eq_support_max' (h : p ≠ 0) :
 (le_max' _ _ $ nat_degree_mem_support_of_nonzero h).antisymm $
   max'_le _ _ _ le_nat_degree_of_mem_supp
 
+lemma degree_eq_support_max (p : R[X]) :
+  p.degree = p.support.max :=
+begin
```

Comment:
Just to check; is this true by rfl?

Reply:
You are right: this is `rfl`!  It might be that it is rfl now, thanks to the refactor that Yakov mentioned.

Do you think that the lemma is still worth having?

Reply:
I think perhaps that degree should be directly defined with this spelling, and then the lemma is just `rw degree`

Reply:
Should I open a Zulip poll about this?  Hasn't the definition of degree just changed?

Reply:
Eric, I've decided to give the refactor a try.  This is an attempt: I want to find out how much effort it will be to change the definition.

The three `finset.max` lemmas are still here, but I also opened PR #15212 containing just those lemmas.